app/controllers/application_controller.rb

```diff
@@ -1,10 +1,28 @@
 class ApplicationController < ActionController::Base
+	before_action :current_user
   protect_from_forgery with: :exception
 
-  helper_method :cart
+  helper_method :cart, :current_user, :current_admin?, :current_merchant?
 
   def cart
     @cart ||= Cart.new(session[:cart] ||= Hash.new(0))
   end
 
+	def current_user
+	 	return false unless session[:user_id]
```

Comment:
Nice guard. I actually think you might be able to one line it with something like:
`@current_user ||= User.find(session[:user_id]) if session[:user_id]`

But that is totally something we can think about later. I actually needed this method for my profile page and had written it out, good to know I did it the right way when I worked on it, haha.